components-java-sdk/src/main/java/io/dapr/components/domain/state/GetResponse.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.domain.state;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Response for a StateStore#get request.
+ *
+ * @param data The value of the GetRequest response.
+ * @param etag The etag of the associated key.
+ * @param metadata Metadata related to the response.
+ * @param contentType The response value contenttype
+ */
+public record GetResponse(byte[] data, String etag, Map<String, String> metadata, String contentType) {
```

Comment:
For types typically instantiated by the developer's code (i.e. responses rather than requests), as many of the properties are optional (e.g. ETag, metadata, etc.), any thoughts on allowing the developer to omit them?  I don't know if Java has similar constructs as .NET to make that convenient.

Reply:
That is a good one. There isn't `= default` or optional parameters in java (I mean, having `Optional<>` around isn't the same). What I can do, and I think it would have the same effect of what you want, is add overloaded constructors that only have required fields.

I can do that and I think it will make DevEx better.

Reply:
Addressed in #16